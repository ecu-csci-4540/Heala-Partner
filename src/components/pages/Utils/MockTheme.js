import React from "react";
import { createTheme } from "@mui/material/styles";
import { ThemeProvider } from "@mui/material/styles";

function MockTheme({ children }) {
  const theme = createTheme({});
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export default MockTheme;
