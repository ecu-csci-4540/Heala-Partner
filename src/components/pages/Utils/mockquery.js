import { Login_USER } from "components/graphQL/Mutation";

export const mockLogin = [
  {
    request: {
      query: Login_USER,
      variables: {
        data: {
          email: "shally@gmail.com",
          password: "admin1234",
          authType: "normal",
        },
      },
    },
    result: {
      data: {
        login: {
          account: {
            _id: "628b4de7161f3d0013b7e195",
            dociId: "HEALA-HL9GALYS",
            email: "shally@gmail.com",
            isEmailVerified: "false",
            access_token:
              "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoic2hhbGx5QGdtYWlsLmNvbSIsImlhdCI6MTY1ODczNjI3OCwiZXhwIjoxNjU4ODIyNjc4LCJhdWQiOiJhdXRoIiwiaXNzIjoiaGVhbGEiLCJzdWIiOiI2MjhiNGRlNzE2MWYzZDAwMTNiN2UxOTUifQ.gSZd68tNKja7fr0tQLzKLztgrGkT1gQ9ZAUvkRtZOe0",
            refresh_token:
              "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoic2hhbGx5QGdtYWlsLmNvbSIsImlhdCI6MTY1ODczNjI3OCwiZXhwIjoxNjU4OTA5MDc4LCJhdWQiOiJhdXRoIiwiaXNzIjoiaGVhbGEiLCJzdWIiOiI2MjhiNGRlNzE2MWYzZDAwMTNiN2UxOTUifQ.pxpjNNSxOunn5PXsBtqAOFid1wJuwoHHgmE24U5wwgU",
            nextStep: null,
            deactivated: "false",
            deactivatedAt: null,
            deactivateType: null,
            isPasswordTemporary: "false",
          },
          errors: null,
        },
      },
    },
  },
];
