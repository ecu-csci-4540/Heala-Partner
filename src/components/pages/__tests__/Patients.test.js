import Patients from "../Patients";
import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { store } from "store";
import "@testing-library/jest-dom/extend-expect";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "react-redux";
import MockTheme from "../Utils/MockTheme";
import { patientsMock } from "../Utils/mockquery";
import wait from "waait";

/* Resources used to learn*/

/* video on testing graphql
 https://www.youtube.com/watch?v=gg-H6yY2xw0
 https://www.youtube.com/watch?v=_X-_X-_X-_X&list=PL-osiE80TeTt2d9bfVyTiXJA-i1sel30w&index=1

 mock provider for multiple queries
 https://spin.atomicobject.com/2022/07/11/multiple-queries-mock-provider/
 https:stackoverflow.com/questions/66455345/apollos-mockedprovider-doenst-work-when-a-component-has-multiple-usequerys
https: github.com/apollographql/apollo-client/issues/5920*/

// mocking the usealert hook
jest.mock("../../hooks/useAlert", () => () => {
  const useAlert = jest.requireActual("../../hooks/useAlert");
  return {
    ...useAlert,
    displayMessage: jest.fn(),
  };
});

beforeEach(async () => {
  render(
    <BrowserRouter>
      <MockedProvider mocks={patientsMock} addTypename={false}>
        <MockTheme>
          <Provider store={store}>
            <Patients />
          </Provider>
        </MockTheme>
      </MockedProvider>
    </BrowserRouter>
  );
});

afterAll(async () => {
  const { container } = render(
    <BrowserRouter>
      <MockedProvider mocks={patientsMock} addTypename={false}>
        <MockTheme>
          <Provider store={store}>
            <Patients />
          </Provider>
        </MockTheme>
      </MockedProvider>
    </BrowserRouter>
  );
  await wait(1);

  expect(container).toMatchSnapshot();
});

//  This test is to check if the component renders correctly
describe("<Patients Page />", () => {
  it("Should render the graphQL data correctly", async () => {
    const x = await screen.findByText("Damilare Heala");
    expect(x).toBeInTheDocument();
  });
  it("ensure the loader is displayed", () => {
    const loader = screen.getByTestId("loader");
    expect(loader).toBeInTheDocument();
  });
  it("should show pagination data", async () => {
    const x = await screen.findByText("Patients per page");
    expect(x).toBeInTheDocument();
  });
  it("Ensure data is consistent", async () => {
    const x = await screen.findByTestId("test-value");
    expect(x).toHaveTextContent("Paelon Memorial Hospital");
  });
});
