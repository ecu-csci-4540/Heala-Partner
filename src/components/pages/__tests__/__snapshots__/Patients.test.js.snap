// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Patients Page /> Ensure data is consistent 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-4epmb1-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 makeStyles-searchFilterContainer-82 css-1tw2i9j-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 css-1lza66c-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-1soyhaz-MuiGrid-root"
        >
          <div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-s24wv3-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-107499w-MuiGrid-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-s24wv3-MuiGrid-root"
                    >
                      
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl makeStyles-select-95 css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                      >
                        <div
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-labelledby="mui-component-select-search"
                          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                          id="mui-component-select-search"
                          role="button"
                          tabindex="0"
                        >
                          By ID
                        </div>
                        <input
                          aria-hidden="true"
                          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                          name="search"
                          tabindex="-1"
                          value="id"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-hdw1oc"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-11ypecn-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-adornedStart css-1ijc0su-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <h5
                  class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                >
                  HEALA-
                </h5>
              </div>
              <input
                aria-label="Search by ID e.g 7NE6ELLO"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                id="input-with-icon-adornment"
                placeholder="Search by ID e.g 7NE6ELLO"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-syi4na-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-102 css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            to=""
            type="submit"
          >
            Search
             
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 css-gt77ug-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 css-1wjyquw-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-s24wv3-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-107499w-MuiGrid-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-colorPrimary makeStyles-FormLabel-94 css-u4tvz2-MuiFormLabel-root"
                    >
                      By Gender
                    </label>
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-s24wv3-MuiGrid-root"
                      >
                        
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl makeStyles-select-95 css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                        >
                          <div
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-labelledby="mui-component-select-gender"
                            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                            id="mui-component-select-gender"
                            role="button"
                            tabindex="0"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                            >
                              None
                            </p>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                            name="gender"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                            data-testid="ArrowDropDownIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-hdw1oc"
                            >
                              <span
                                class="notranslate"
                              >
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-1ypag5h-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-8atqhb"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-k4xeig-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-2pbtu6-MuiTypography-root"
              id="tableTitle"
            />
          </div>
          <div
            class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
          >
            <table
              aria-labelledby="tableTitle"
              class="MuiTable-root css-1gplja-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root makeStyles-tableHeader-98 css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-hbtnrh-MuiTableCell-root"
                    scope="col"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        aria-label="select all desserts"
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    User ID
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Patient Name
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Plan
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Provider
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Consultations
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Status
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableHeaderCell-99 css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  aria-checked="false"
                  class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
                  role="checkbox"
                  tabindex="-1"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-1nf681w-MuiTableCell-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        aria-labelledby="enhanced-table-checkbox-0"
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableCell-84 css-1ex1afd-MuiTableCell-root"
                    id="enhanced-table-checkbox-0"
                    scope="row"
                    style="text-align: left;"
                  >
                    EBDASEHA
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableCell-84 css-1ex1afd-MuiTableCell-root"
                  >
                    <div
                      style="height: 100%; display: flex; align-items: center;"
                    >
                      <span
                        style="margin-right: 1rem;"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circular css-7g00h4-MuiAvatar-root"
                        >
                          <img
                            alt="Display Photo of Damilare"
                            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                            src="avatar.svg"
                          />
                        </div>
                      </span>
                      <span
                        style="font-size: 1.25rem;"
                        test="test-value"
                      >
                        Damilare Heala
                      </span>
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableCell-84 css-1ex1afd-MuiTableCell-root"
                  >
                    Basic
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableCell-84 css-1ex1afd-MuiTableCell-root"
                    data-testid="test-value"
                  >
                    Paelon Memorial Hospital
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableCell-84 css-1ex1afd-MuiTableCell-root"
                  >
                    0
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-tableCell-84 css-1ex1afd-MuiTableCell-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault makeStyles-badge-85 css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        Active
                      </span>
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  >
                    <a
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-83 css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="/patients/62e14eb149003e0013e02895"
                      tabindex="0"
                    >
                      View Profile
                      <span
                        class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowForwardIosIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="MuiTablePagination-root makeStyles-pagination-103 css-jtlhu6-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id="mui-8"
              >
                Patients per page
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                variant="standard"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-8 mui-7"
                  class="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                  id="mui-7"
                  role="button"
                  tabindex="0"
                >
                  10
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="10"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                1–6 of 6
              </p>
              <div
                class="MuiBox-root css-1zye22"
              >
                <button
                  aria-label="first page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FirstPageIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="last page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LastPageIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-nen11g-MuiStack-root"
  />
</div>
`;
